cEP-0026.md

```diff
@@ -0,0 +1,189 @@
+# Optimize caching for the NextGen-Core
+
+| Metadata |                                           |
+| ---------| ----------------------------------------- |
+| cEP      | 26                                        |
+| Version  | 1.0                                       |
+| Title    | Optimize caching for the NextGen-Core     |
+| Authors  | Palash Nigam <mailto:npalash25@gmail.com> |
+| Status   | Implementation Due                        |
+| Type     | Feature                                   |
+
+## Abstract
+
+This document describes how a caching mechanism (an enhanced version of the one
+being used by the old core) will be implemented and integrated with the
+NextGen-Core as a part of this
+[GSoC project](https://summerofcode.withgoogle.com/projects/#6434190552203264).
+
+## Introduction
+
+Since the NextGen-Core aims to provide faster speed and better performance for
+coala-runs on large codebases it needs caching to achieve this kind of
+performance.
+
+The current caching used by the old core can be improved upon and integrated
+with the NextGen-Core by adding a few more enhancements to the current caching
```

Comment:
> integrated with the NextGen-Core

you can integrate the old stuff into the new one? I'm wondering :D

Reply:
That's not exactly what I meant. What I meant was that we try to take the improvements suggested for the old cache and try to integrate it with the new cache by putting a little spin on it. I can see how that can be confusing.

I now realise that its not important to include that, its confusing. Deleting it.